tests/data/test_generate.py

```diff
@@ -1,43 +1,139 @@
 import pytest
+from expertsystem.amplitude.model import AmplitudeModel
 
-from tensorwaves.data.generate import generate_phsp
+from tensorwaves.data.generate import generate_data, generate_phsp
+from tensorwaves.data.tf_phasespace import TFUniformRealNumberGenerator
+from tensorwaves.physics.helicity_formalism.amplitude import IntensityBuilder
 from tensorwaves.physics.helicity_formalism.kinematics import (
     HelicityKinematics,
     ParticleReactionKinematicsInfo,
 )
 
 
+def test_generate_data(helicity_model: AmplitudeModel):
+    model = helicity_model
+    kinematics = HelicityKinematics.from_model(model)
+    phsp_sample = generate_phsp(100, kinematics)
+    builder = IntensityBuilder(model.particles, kinematics, phsp_sample)
+    intensity = builder.create_intensity(model)
+    sample_size = 3
+    rng = TFUniformRealNumberGenerator(seed=0)
+    sample = generate_data(
+        sample_size, kinematics, intensity, random_generator=rng
+    )
```

Comment:
So, what happens here is a bit tricky, as rightly pointed out https://github.com/ComPWA/tensorwaves/issues/59#issuecomment-716658526: There's just one seed, and one generator that's used by both `generate_data` and `generate_phsp`.